docs/csharp/programming-guide/main-and-command-args/main-return-values.md

```diff
@@ -33,58 +33,91 @@ translation.priority.ht:
- [How to: Display Command Line Arguments](../../../csharp/programming-guide/main-and-command-args/how-to-display-command-line-arguments.md)   
- [How to: Access Command-Line Arguments Using foreach](../../../csharp/programming-guide/main-and-command-args/how-to-access-command-line-arguments-using-foreach.md)
+
+# Main() return values (C# Programming Guide)
+
+The `Main` method can return `void`:
+
+[!code-cs[csProgGuideMain#12](../../../csharp/programming-guide/inside-a-program/codesnippet/CSharp/main-return-values_1.cs)]
+
+It can also return an `int`:
+
+!code-cs[csProgGuideMain#13](../../../csharp/programming-guide/inside-a-program/codesnippet/CSharp/main-return-values_2.cs)]
+
+If the return value from `Main` is not used, returning `void` allows for slightly simpler code. However, returning an integer enables the program to communicate status information to other programs or scripts that invoke the executable file. The following example shows how the return value from `Main` can be accessed.
+
+## Example
+
+This example uses [.NET Core](../../../core/index.md) command line tools. If you are unfamilar with .NET Core command line tools, you can learn about them in this [Get started topic](../../../core/tutorials/using-with-xplat-cli.md).
+
+Modify the `Main` method in *program.cs* as follows:
+
+[!code-cs[csProgGuideMain#14](../../../csharp/programming-guide/inside-a-program/codesnippet/CSharp/main-return-values_3.cs)]  
+
+When a program is executed in Windows, any value returned from the `Main` function is stored in an environment variable. This environment variable can be retrieved using `ERRORLEVEL` from a batch file, or from `$LastExitCode` from powershell.
+
+You can build the application using the [dotnet CLI](../../../core/tools/dotnet.md) `dotnet build` command.
+
+Next, create a Powershell script to run the application and display the result. Paste the following code into a text file and save it as `test.ps1` in the folder that contains the project. Run the powershell script by typing `test.ps1` at the powershell prompt.
+
+Because the code returns zero, the batch file will report success. However, if you change MainReturnValTest.cs to return a non-zero value and then re-compile the program, subsequent execution of the powershell script will report failure.
+
+```powershell
+dotnet run
+if ($LastExitCode -eq 0) {
+    Write-Host "Execution succeeded"
+} else
+{
+    Write-Host "Execution Failed"
+}
+Write-Host "Return value = " $LastExitCode
+```
+
+## Sample output
+
+```txt
+Execution succeeded
+Return value = 0
+```
+
+## Async Main return values
+
+Async Main return values move the boilerplate code necessary for calling asynchronous methods in `Main` to code generated by the compiler. Previously, you would need to write this construct to call asynchronous code and ensure your program ran until the asynchronous operation completed:
+
+```csharp
+public static void Main()
```

Comment:
The doc earlier explicitly stated that Main shouldn't be public (and I commented on why it was making that guidance), but here it's public :smile: